### PR TITLE
Voting 2024_04_23 (GateSeal)

### PR DESCRIPTION
**Pull request by mashahappyholly**
added vote and test scripts
added Escrow, SealableMock, VotingAdapter in interfaces

### CODE REVIEW

File: interfaces/Escrow.json
Comment (zuzueeka) [inline interfaces/Escrow.json:1]: do we need this file for this vote?

File: interfaces/SealableMock.json
Comment (zuzueeka) [inline interfaces/SealableMock.json:1]: seems that it isn't a mock

File: interfaces/VotingAdapter.json
Comment (zuzueeka) [inline interfaces/VotingAdapter.json:1]: I think it's not needed

File: scripts/vote_2024_04_23.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    get_is_live,
+    get_priority_fee,
+    contracts,
+)
+from utils.agent import agent_forward
+from utils.permissions import encode_oz_grant_role, encode_oz_revoke_role
+
+proposed_gate_seal_address = "0x79243345eDbe01A7E42EDfF5900156700d22611c"
+
+description = """
```
Comment (zuzueeka) [inline scripts/vote_2024_04_23.py:28]: all links should be presented as [...](...) due to markdown marking

File: scripts/vote_2024_04_23.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                [
+                    encode_oz_grant_role(
+                        contract=contracts.withdrawal_queue,
+                        role_name="PAUSE_ROLE",
+                        grant_to=proposed_gate_seal_address,
+                    )
+                ]
+            ),
+        ),
+
```
Comment (zuzueeka) [inline scripts/vote_2024_04_23.py:55]: redundant spase

File: tests/snapshot/test_voting.py [hunk 3]
```diff
@@ -106,4 +106,4 @@ def test_create_wait_enact(helpers, vote_time, call_target, vote_ids_from_env):
             assert_expected_diffs(
                 step_name, diff, {"votesLength": ValueChanged(from_val=votesLength + 1, to_val=votesLength + 2)}
             )
-        assert_no_diffs(step_name, diff)
+        assert_no_diffs(step_name, diff)
```
Comment (zuzueeka) [inline tests/snapshot/test_voting.py:109]: it looks like an extra space or something like formatting, it's better if there is no diff

File: tests/test_2024_04_23.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+new_gate_seal = "0x79243345eDbe01A7E42EDfF5900156700d22611c"
+
+def _check_role(contract: Contract, role: str, holder: str):
+    role_bytes = web3.keccak(text=role).hex()
+    assert contract.getRoleMemberCount(role_bytes) == 1, f"Role {role} on {contract} should have exactly one holder"
+    assert contract.getRoleMember(role_bytes, 0) == holder, f"Role {role} holder on {contract} should be {holder}"
+
+def _check_no_role(contract: Contract, role: str, holder: str):
+    role_bytes = web3.keccak(text=role).hex()
+    assert contract.getRoleMemberCount(role_bytes) == 1, f"Role {role} on {contract} should have exactly one holder"
```
Comment (zuzueeka) [inline tests/test_2024_04_23.py:35]: it was my bad, better to remove this check 'contract.getRoleMemberCount(role_bytes) == 1,'

File: tests/test_2024_04_23.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    #Асceptance test
+    new_gate_seal_contract = interface.GateSeal(new_gate_seal)
+    assert new_gate_seal_contract.get_sealing_committee() == GATE_SEAL_COMMITTEE
+    sealables = new_gate_seal_contract.get_sealables()
+    assert len(sealables) == 2
+    assert contracts.validators_exit_bus_oracle.address in sealables
+    assert contracts.withdrawal_queue.address in sealables
+
+    assert new_gate_seal_contract.get_seal_duration_seconds() == GATE_SEAL_PAUSE_DURATION
+    assert new_gate_seal_contract.get_expiry_timestamp() == 1743465600
```
Comment (zuzueeka) [inline tests/test_2024_04_23.py:77]: please, assign a separate variable in the beginning:
EXPIRY_TIMESTAMP =  1743465600 # Tue Apr 01 2025 00:00:00 GMT+0000

File: tests/test_2024_04_23.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    expiry_timestamp = chain.time()
+    assert new_gate_seal_contract.is_expired()
+    assert expiry_timestamp - new_gate_seal_contract.get_expiry_timestamp() <= 2
+    print("Expired")
+
+    for sealable in sealables:
+        assert interface.IPausable(sealable).isPaused()
+    print("Sealables paused")
+
+    chain.sleep(6 * 60 * 60 * 24+100)
```
Comment (zuzueeka) [inline tests/test_2024_04_23.py:95]: 6 * 60 * 60 * 24+100 -> 6 * 60 * 60 * 24 + 100

File: tests/snapshot/test_voting.py (hunk 3 above)
Comment (mashahappyholly) [reply tests/snapshot/test_voting.py:109]: @zuzueeka There are really no changes here. I accidentally moved something, then put it back the way it was

File: tests/test_2024_04_23.py (hunk 4 above)
Comment (mashahappyholly) [reply tests/test_2024_04_23.py:35]: agreed to delete

**Review by zuzueeka — APPROVED** [review]
LGTM

**Pull request merged**